lib/TPP/ConvertCheckToLoops.cpp

```diff
@@ -107,8 +106,70 @@ struct ConvertExpectTrueOp : public OpRewritePattern<ExpectTrueOp> {
   }
 };
 
+// Convert
+// check.expect_sane(%t1:tensor<2x2xf32>)
+// to the pattern:
+// scf.some_loop(%i, %j)
+// %0 = load from %t1
+// %1 = arith.cmpf ord, %0, %nan : f32
+// %2 = arith.cmpf one, %0, %inf : f32
+// %compare = arith.andi %1, %2 : i1
+// cf.assert %compare, "Buffer can't contain NaNs or Infinite values"
+struct ConvertExpectSaneOp : public OpRewritePattern<ExpectSaneOp> {
+  using OpRewritePattern<ExpectSaneOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(ExpectSaneOp expectSaneOp,
+                                PatternRewriter &rewriter) const override {
+    Location loc = expectSaneOp.getLoc();
+    auto elementType = expectSaneOp.getOperand().getType().getElementType();
+    SmallVector<Value> ubs;
+    if (!expectSaneOp.getOperand().getType().isa<MemRefType>() ||
+        !elementType.isa<mlir::FloatType>()) {
+      return failure();
+    }
+    size_t rank =
+        expectSaneOp.getOperand().getType().cast<MemRefType>().getRank();
+    for (size_t idx = 0; idx < rank; idx++) {
+      auto dim = linalg::createOrFoldDimOp(rewriter, loc,
+                                           expectSaneOp.getOperand(), idx);
+      ubs.push_back(dim);
+    }
+    Value zero = rewriter.create<arith::ConstantIndexOp>(loc, 0);
+    SmallVector<Value> lbs(rank, zero);
+    Value one = rewriter.create<arith::ConstantIndexOp>(loc, 1);
+    SmallVector<Value> steps(rank, one);
+    (void)scf::buildLoopNest(
+        rewriter, loc, lbs, ubs, steps,
+        [&](OpBuilder &b, Location loc, ValueRange localIvs) {
+          Value scalarOperand = b.create<memref::LoadOp>(
+              loc, expectSaneOp.getOperand(), localIvs);
+          Value zeroVal = b.create<arith::ConstantOp>(
+              loc, elementType, rewriter.getZeroAttr(elementType));
+
+          Value inf = rewriter.create<arith::ConstantOp>(
```

Comment:
what about `-Inf`?

Reply:
Modifying the check to compare against absolute value of the operand.